src/BloomExe/Program.cs

```diff
@@ -172,11 +171,20 @@ static void Main(string[] args1)
 							InstallerSupport.MakeBloomRegistryEntries();
 							Browser.SetUpXulRunner();
 							Browser.XulRunnerShutdown += OnXulRunnerShutdown;
-							L10NSharp.LocalizationManager.SetUILanguage(Settings.Default.UserInterfaceLanguage, false);
+							LocalizationManager.SetUILanguage(Settings.Default.UserInterfaceLanguage, false);
 							var transfer = new BookTransfer(new BloomParseClient(), ProjectContext.CreateBloomS3Client(),
 								_applicationContainer.HtmlThumbnailer, new BookDownloadStartingEvent())/*not hooked to anything*/;
 							transfer.HandleBloomBookOrder(args[0]);
 							PathToBookDownloadedAtStartup = transfer.LastBookDownloadedPath;
+
+							// BL-2143: Show failure message if download was not successful
+							if (string.IsNullOrEmpty(PathToBookDownloadedAtStartup))
+							{
+								var errCaption = LocalizationManager.GetString("Download.FailedCaption", "Download not completed");
+								var errMessage = LocalizationManager.GetString("Download.Failed", "Your download from BloomLibrary.org did not complete.");
+								MessageBox.Show(errMessage, errCaption, MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
+							}
```

Comment:
Should be fairly easy to test...something is bound to go wrong if you unplug the cable during the download.